sql/sql_show.cc

```diff
@@ -9226,18 +9226,12 @@ static int make_slave_status_old_format(THD *thd, ST_SCHEMA_TABLE *schema_table)
   for (uint i=0; !field_info->end_marker(); field_info++, i++)
   {
     if (all_slaves ||
-        // not SLAVE_STATUS_COL_CONNECTION_NAME,
-        // SLAVE_STATUS_COL_SLAVE_SQL_STATE
-        // and less
-        // SLAVE_STATUS_COL_RETRIED_TRANSACTIONS
-        !(i <= SLAVE_STATUS_COL_SLAVE_SQL_STATE ||
-          i >= SLAVE_STATUS_COL_RETRIED_TRANSACTIONS))
+        (i > SLAVE_STATUS_COL_SLAVE_SQL_STATE &&
+         i < SLAVE_STATUS_COL_RETRIED_TRANSACTIONS))
     {
       LEX_CSTRING field_name= field_info->name();
       Item_field *field= new (thd->mem_root)
         Item_field(thd, context, field_name);
-      DBUG_ASSERT(all_slaves || (i > SLAVE_STATUS_COL_SLAVE_SQL_STATE &&
```

Comment:
Hmm... I recall some funkiness here when the `DBUG_ASSERT` was added. I agree it does nothing right now, but I wonder if somewhere along the way, the intended assertion was lost along the way, and we can correct it (rather than remove it)

Reply:
https://github.com/MariaDB/server/commit/25b5c63905682a0e38d8ed86170d98be6ddb8cf8#diff-cd5ac2db946c26a2668c92b5804888e2e68724aaee6b1a86c255e2271ef3f835R9140-R9141
I think it’s already corrected when the `if`’s condition switched from a bitfield.